server/context.go

```diff
@@ -204,6 +211,18 @@ func (ctx *Context) parseGossipBootstrapResolvers() ([]gossip.Resolver, error) {
 	return bootstrapResolvers, nil
 }
 
+// GetHTTPClient returns the context http client, initializing it
+// if needed. It uses the context Certs.
+// TODO(marc): are there odd cases where may need a mutex?
+// In the worst case, we'll be creating more than one.
+func (ctx *Context) GetHTTPClient() (*http.Client, error) {
+	var err error
+	if ctx.HTTPClient == nil {
+		ctx.HTTPClient, err = client.NewHTTPClient(ctx.Certs)
+	}
+	return ctx.HTTPClient, err
```

Comment:
I suppose that's the question. I haven't seen any cases where we may issue more multiple requests in parallel, and even if we do, we'll just create multiple clients. If we're worried, we can always add a mutex.